corehq/ex-submodules/casexml/apps/phone/data_providers/case/livequery.py

```diff
@@ -58,6 +58,51 @@ def enliven(case_id):
             for next_id in chain(ext_ids, host_ids):
                 enliven(next_id)
 
+    def classify(index):
+        """Classify index as either live or extension with live status pending
+
+        This closuer mutates `next_ids`, `extensions_by_host`,
```

Comment:
sometimes closures can be replaced with helper classes (in the case `do_livequery(*args, **kwargs)` would be a wrapper like `return LiveQueryHelper(*args, **kwargs).main()`, and `next_ids` would then be `self.next_ids`), but that doesn't always end up feeling more elegant/easier to reason with. Since there are a few inner functions here and they're somewhat long and involved, I thought I'd at least suggest it and leave it up to you to decide whether it would help with organization or not.